### ISSUE TITLE
Navigation: Improve `MegaMenu` a11y

### ISSUE THREAD

**Issue by ashharrison90** [grafana/grafana#77263]
currently `MegaMenu` is keyboard accessible via the tab key. this works great in an overlay, but might be annoying when docked since you have to tab through all the elements in the menu. we do have some mechanism to help via the `Skip to main content` link, but maybe this isn't enough? 

TODO: 
- [x] investigate docked sidebars in other products - what do they do? 
  - e.g. [github file search](https://github.com/grafana/grafana?search=1) uses arrow keys to navigate
  - e.g. spotify artists sidebar
- [x] check which html tags are currently in use for the nav (<nav> can be skipped when using a screenreader)
- [x] #80020
  - Megamenu is a nav element, but it's not announced as such
  - Between hamburger menu button and the actual nav there are two other elements
  - When entering the MegaMenu the number of items in the main unordered list is not announced 
- [x] check screen reader is announcing things correctly

**Comment by L-M-K-B** [comment]
I would be interested in working on this. @ashharrison90 let's talk about it next week.
